e2e/dispute_test.go

```diff
@@ -131,95 +110,45 @@ func testDisputeTransitionC2T(
 	client jsonrpc.RPCClient,
 	ethClient *eth.Client,
 	senderWallet bls.Wallet,
+	receiverWallet bls.Wallet,
 	wallets []bls.Wallet,
 ) {
-	sink := make(chan *rollup.RollupRollbackStatus)
-	subscription, err := ethClient.Rollup.WatchRollbackStatus(&bind.WatchOpts{}, sink)
-	require.NoError(t, err)
-	defer subscription.Unsubscribe()
-
-	firstC2TWallet := wallets[len(wallets)-32]
-	sendC2TBatchWithInvalidAmount(t, ethClient, firstC2TWallet.PublicKey())
-	testRollbackCompletion(t, ethClient, sink, subscription)
+	requireRollbackCompleted(t, ethClient, func() {
+		sendC2TBatchWithInvalidStateRoot(t, ethClient, receiverWallet.PublicKey(), 3)
+	})
 
-	testBatchesAfterDispute(t, client, 3)
+	requireBatchesCount(t, client, 3)
 
+	firstC2TWallet := wallets[len(wallets)-32]
 	submitTxBatchAndWait(t, client, func() common.Hash {
 		return testSubmitC2TBatch(t, client, senderWallet, wallets, firstC2TWallet.PublicKey(), 64)
 	})
 }
 
-func testDisputeTransitionMM(
-	t *testing.T,
-	client jsonrpc.RPCClient,
-	ethClient *eth.Client,
-	senderWallet bls.Wallet,
-) {
-	sink := make(chan *rollup.RollupRollbackStatus)
-	subscription, err := ethClient.Rollup.WatchRollbackStatus(&bind.WatchOpts{}, sink)
-	require.NoError(t, err)
-	defer subscription.Unsubscribe()
-
-	sendMMBatchWithInvalidAmount(t, ethClient)
-	testRollbackCompletion(t, ethClient, sink, subscription)
-
-	testBatchesAfterDispute(t, client, 4)
-
-	submitTxBatchAndWait(t, client, func() common.Hash {
-		return testSubmitMassMigrationBatch(t, client, senderWallet, 96)
+func testDisputeTransitionMM(t *testing.T, client jsonrpc.RPCClient, ethClient *eth.Client) {
+	requireRollbackCompleted(t, ethClient, func() {
+		sendMMBatchWithInvalidStateRoot(t, ethClient, 4)
 	})
-}
-
-func testDisputeTransitionTransferInvalidStateRoot(t *testing.T, client jsonrpc.RPCClient, ethClient *eth.Client) {
-	sink := make(chan *rollup.RollupRollbackStatus)
-	subscription, err := ethClient.Rollup.WatchRollbackStatus(&bind.WatchOpts{}, sink)
-	require.NoError(t, err)
-	defer subscription.Unsubscribe()
 
-	sendTransferBatchWithInvalidStateRoot(t, ethClient)
-	testRollbackCompletion(t, ethClient, sink, subscription)
-
-	testBatchesAfterDispute(t, client, 5)
+	requireBatchesCount(t, client, 4)
```

Comment:
Why not send valid batches before and after the incorrect one the same as in `testDisputeTransitionTransfer`?

Reply:
I guess the purpose of sending valid batches in between invalid ones is to check that we can produce another valid batch after a rollback. 

If this is the case sending just a single batch in between invalid batches should be fine. I've added sending a valid MM batch after a MM batch rollback.